packages/icon-button/src/FzIconButton.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <button
+    type="button"
+    class="fz__iconbtn relative"
+    :disabled="disabled"
+    :class="classes"> 
+      <fz-icon 
+        class="fz__iconbtn__icon"
+        :name="iconName"
+        :variant="iconVariant"
+        :size="mappedIconSize"/>
+      <div class="fz__iconbtn__notification" 
+        v-if="variant === IconButtonVariant.notification"
+        :class="notificationClasses">
+      </div>
+      <span class="hidden w-0 h-0">{{ tooltip }}</span>
+    </button>
+</template>
+
+<script lang="ts" setup>
+import {computed} from 'vue'
+import {IconButtonVariant} from './types';
+import {FzIcon } from '@fiscozen/icons';
+import type { IconVariants } from '@fiscozen/icons';
+
+const props = withDefaults(
+  defineProps<{
+    /**
+     * The tooltip of the button will be shown on hover
+     */
+    tooltip: string
+    /**
+     * primary or secondary button
+     */
+    variant?: IconButtonVariant
+    /**
+     * size of the button
+     */
+    size?: 'sm' | 'md' | 'lg',
+    /**
+     * whether action is enabled or not
+     */
+    disabled: boolean,
+    /**
+     * Icon to be displayed. Use fontawesome search here https://fontawesome.com/v6/icons
+     * Mind that not all of the icons and variants might be available.
+     */
+    iconName: string,
+    /**
+     * Fontawesome icon variant: solid, regular, light, thin. Sharp subvariants are available as well
+     */
+    iconVariant?: IconVariants,
```

Comment:
C'è inconsistenza nei separatori dell'interfaccia, (`,`), eslint non fa enforce di questa cosa al momento?

Reply:
in effetti curiosamente no, good catch. Controllo

Reply:
ok, vista lo shitshow riguardo alla migrazione delle regole di formattazione di eslint direi che rimando questa cosa, ragioniamoci un attimo e vediamo cosa fare e quali tool esattamente vogliamo tenere